V/VTKMinimal/build_tarballs.jl

```diff
@@ -0,0 +1,128 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "VTKMinimal"
+version = v"8.2.0"
+
+# Build a minimal subset of VTK sufficient for rendering purposes. Notably, this leaves
+# out HDF5 and MPI, thus creating fewer clash points for transient linking problems.
+# Building the full version on BinaryBuilder seems to be feasible, though.      
+#
+# The platform constraint to linux + FreeBSD comes from the constraints on libXt etc.
+# Currently, this depends on the Xorg libraries, therefore it is not available for Mac+Win,
+# this situation may change though with Qt.
+#
+# A couple of libraries (eigen, doubleconversion, ogg, theora) is used in the
+# vtk built-in version. It appears to be possible to replace them by jlls upon availability of those.
+# Though there is Ogg_jll, the vtktheora does not go well with it.
+#
+#
+
+# Grab the source directly from the vtk website
+sources = [
+    "https://www.vtk.org/files/release/8.2/VTK-8.2.0.tar.gz" => "34c3dc775261be5e45a8049155f7228b6bd668106c72a3c435d95730d17d57bb"
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+EXTRA_VARS=()
+if [[ "${target}" == *-linux-* ]] || [[ "${target}" == *-freebsd* ]]; then
+    # On Linux and FreeBSD this variable by default does `-L/usr/lib`
+    EXTRA_VARS+=(LDFLAGS.EXTRA="")
+fi
+
+
+mkdir build
+cd build
+cmake -DVTK_CUSTOM_LIBRARY_SUFFIX="" -DVTK_USE_SYSTEM_LIBRARIES:BOOL=ON -DVTK_USE_SYSTEM_EIGEN3:BOOL=OFF  -DVTK_USE_SYSTEM_THEORA:BOOL=OFF   -DVTK_USE_SYSTEM_EIGEN=OFF -DVTK_USE_SYSTEM_DOUBLECONVERSION=OFF  -DVTK_USE_SYSTEM_LZMA:BOOL=ON -DVTK_USE_SYSTEM_OGG:BOOL=OFF   -DVTK_USE_SYSTEM_PNG:BOOL=ON  -DVTK_USE_SYSTEM_LZ4=ON -DVTK_USE_SYSTEM_JPEG:BOOL=ON -DVTK_USE_SYSTEM_TIFF:BOOL=ON -DVTK_USE_SYSTEM_ZLIB:BOOL=ON -DVTK_Group_Rendering:BOOL=OFF -DVTK_Group_StandAlone:BOOL=OFF -DModule_vtkCommonCore:BOOL=ON -DModule_vtkCommonDataModel:BOOL=ON -DModule_vtkCommonExecutionModel:BOOL=ON -DModule_vtkCommonMisc:BOOL=ON -DModule_vtkCommonSystem:BOOL=ON -DModule_vtkCommonMath:BOOL=ON -DModule_vtkCommonTransforms:BOOL=ON -DModule_vtkRenderingCore:BOOL=ON -DModule_vtkRenderingContext2D:BOOL=ON -DModule_vtkFiltersCore:BOOL=ON -DModule_vtkFiltersGeneral:BOOL=ON -DModule_vtkFiltersGeometry:BOOL=ON -DModule_vtkFiltersExtraction:BOOL=ON -DModule_vtkFiltersFlowPaths:BOOL=ON -DModule_vtkInteractionWidgets:BOOL=ON -DModule_vtkIOMovie:BOOL=ON -DModule_vtkParallelCore:BOOL=ON -DModule_vtkInteractionStyle:BOOL=ON -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_BUILD_TYPE=Release  ../VTK-8.2.0/
```

Comment:
You should pass also the toolchain file, see the [build tips](https://github.com/JuliaPackaging/Yggdrasil/wiki/Tips-for-Building-Packages#cmake-builds):
```
cmake -DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release
```

Reply:
.. vtk complains about missing vtk compile tools then (I think that's why I removed that flag
anyway, and then forgot).

Supporting this appearantly was introduced shortly after the 8.2 release.
https://gitlab.kitware.com/vtk/vtk/-/commit/d67d1059ac2e2da69e5e0b5543a7c2d6808160d0

So we would get this only from git. 

They say that 9.0.0-rc1 will be out soon. So we might start to deploy for 64bit only and make
an update later ?

Reply:
Edit: hopeful to build now with 9.0.0-rc1

Reply:
... Hopes crashed: VTK builds, but need to learn how to export the xcompilation tools built by vtk to client codes